compiler/z/codegen/BinaryEvaluator.cpp

```diff
@@ -658,34 +632,25 @@ generic32BitSubEvaluator(TR::Node * node, TR::CodeGenerator * cg)
 
       switch (secondChild->getDataType())
          {
-         case TR::Address:
-            TR_ASSERT( TR::Compiler->target.is32Bit(), "generic32BitSubEvaluator(): unexpected data type for 64bit code-gen!");
-         case TR::Int32:
+         case TR::Int8:
```

Comment:
Not necessary for this change, but given this code seems to be a duplicate of the code above it might be nice to move it to its own function.

Reply:
Opened #3668.